### ISSUE TITLE
Attributes on data array in dataset. (Feature)

### ISSUE THREAD

**Issue by shaunc** [astropenguin/xarray-dataclasses#47]
Currently there seems to be no way to specify attributes for a DataArray in a Dataset. Indeed, even if they are passed to `new()` in a DataArray, they are discarded.

It would be nice to reuse DataArray specs to be able specify these:

```python

from xarray_datasetclasses import dataarrayclass, datasetspce, Data, Attr, DataArray

@dataarrayclass
class FooSpec:
    data: Data['x', float]
    meta: Attr[int]

@datasetclass
class BarSpec:
    array: DataArray[FooSpec]

```
Does that look useful? Here `DataArray` is from `xarray_datasetclasses` ... if this name isn't good because of conflict with xarray, perhaps ... `ArraySpec`?

**Comment by astropenguin** [comment]
@shaunc Please let me close the issue because the feature was implemented by #59 and is now available in v0.4.0.
To achieve the feature, I added new type hints (`Dataof` and `Coordof`; see also "Advanced usage" in README).
Please feel free to raise any issues if you have some requests on the feature. Thank you!

**Comment by shaunc** [comment]
Sounds good! (Sorry I haven't been of more help -- swamped by other work!)
